lib/restify-plugins/curl-user-agent.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2012 Mark Cavage, Inc.  All rights reserved.
+// Copyright (c) 2016, Joyent, Inc.
+
+// This is basically the user agent pre-route handler from
+// https://github.com/restify/plugins with a bug fix that was submitted upstream
+// at https://github.com/restify/plugins/pull/40. If and when
+// https://github.com/restify/plugins/pull/40 is merged, the goal is to backport
+// it to whatever restify branch VMAPI uses at that time and remove this local
+// copy.
+
+var assert = require('assert-plus');
+
+/**
+ * This basically exists for curl.  curl on HEAD requests usually
+ * just sits there and hangs, unless you explicitly set
+ * Connection:close.  And in general, you probably want to set
+ * Connection: close to curl anyway.
```

Comment:
Would it make sense to add a test that explicitly ensures things still work fine when the curl client *has* passed 'Connection: close'? Maybe that's going too far? Is there any other possible value for 'Connection: ' that a client might ever set that we're silently overriding? Does that matter?